docs/configuration.md

```diff
@@ -0,0 +1,297 @@
+          slotDuration:      7000
+          maxBlockSize:      2000000
+          maxHeaderSize:     2000000
+          maxTxSize:         4096 # 4 Kb
+          maxProposalSize:   700 # 700 bytes
+          mpcThd:            0.01 # 1% of stake
+          heavyDelThd:       0.005 # 0.5% of stake
+          updateVoteThd:     0.001 # 0.1% of total stake
+          updateProposalThd: 0.1 # 10% of total stake
+          updateImplicit:    10 # slots
+          softforkRule:
+            initThd:        0.9 # 90% of total stake
+            minThd:         0.6 # 60% of total stake
+            thdDecrement:   0.05 # 5% of total stake
+          txFeePolicy:
+            txSizeLinear:
+              a: 155381 # absolute minimal fees per transaction
+              b: 43.946 # additional minimal fees per byte of transaction size
+          unlockStakeEpoch: 18446744073709551615 # last epoch (maxBound @Word64)
+        protocolConstants: &dev_core_genesis_spec_protocolConstants
+          k: 2
+          protocolMagic: 55550001
+          vssMinTTL: 2
+          vssMaxTTL: 6
+        ftsSeed: "c2tvdm9yb2RhIEdndXJkYSBib3JvZGEgcHJvdm9kYSA="
+        heavyDelegation: {}
+        avvmDistr: {}
+```
+
+### Initializer
+
+There are two substantially different types of genesis spec, with the
+difference in `initializer` value. Initializer can be either
+`testnetInitializer` or `mainnetInitializer`.
+
+The key point of `testnetInitializer` is that it contains seed which
+is used to generate secret keys. It's quite convenient for testnet,
+because there we want everyone to be able to have some coins. So we
+can deterministically generate a lot of addresses with some coins and
+let people use them. It's also convenient for devnet clusters
+(i. e. clusters primarily used by developers to test something),
+because in this case we also want to know all secrets. An example of
+`testnetInitializer` was provided above. The most interesting part
+there is `distribution` field:
+
+```
+  distribution:
+   testnetRichmenStakeDistr: []
+```
+
+It means that generated richmen will also be bootstrap stakeholders
+and will be participating in SSC. Note that richmen's keys are
+deterministically generated, so basically everyone has full control
+over system in this case. It's suitable for devnet where we don't care
+about adversaries, but it's not suitable for testnet, because we don't
+want to give full control over the system to arbitrary people.
+
+For testnet there is another possible value of `distribution`:
+`testnetCustomStakeDistr`. It allows one to explicitly specify
+bootstrap stakeholders and VSS certificates. Example:
+
+```
+  distribution:
+    testnetCustomStakeDistr:
+      bootStakeholders:
+        75fc2050ea497eb615461d01170679d143fb70c07c0baf3db54c6237: 1
+        eaba957b871c4d5d9f5eab1a4183a037b3b0a59e31a52185d37627f1: 1
+      vssCerts:
+        75fc2050ea497eb615461d01170679d143fb70c07c0baf3db54c6237:
+          expiryEpoch: 5
```

Comment:
What does `expiryEpoch` mean?

Reply:
It's described at https://github.com/input-output-hk/cardano-sl/tree/cardano-sl-1.0/scripts/prepare-genesis#32-genesis-json-format-explanation as:

> index of epoch when this certificate will expire

@gromakovsky, judiciously linking to parts of the existing genesis generation doc or deprecating it and integrating any other context it provides into this doc may help.

Reply:
@gromakovsky, bump.

Reply:
See `Genesis data format` section, I described it there (basically copied from the old doc).

As I wrote in the first message in this PR, I suggest that other activities related to the old documentation (e. g. merging it with this one) will be performed later, because it's not crucial now.